docs/dev/handbook/cluster_creation.md

```diff
@@ -0,0 +1,187 @@
+Creating clusters with cluster-class
+========================================
+
+This developer handbook describe how to create a cluster using a clusterclass based CRD. 
+
+
+## Set up and host a package repository for management cluster
+First, make sure you have the push access to an OCI registry. Export its URL as `OCI_REGISTRY` environment variable.
+
+### Set up an OCI Registry with GCP
+
+For example, let's suppose we have a project with ID `my-project-1527816345739` on GCP. Then we can set the registry URL
+as `gcr.io/my-project-1527816345739/tkg/management`. Please refer to the
+[GCP Registry documentation](https://cloud.google.com/container-registry/docs/overview) to decide the actual
+URL.
+
+Export the registry URL as `OCI_REGISTRY`
+
+```shell
+export OCI_REGISTRY=gcr.io/my-project-1527816345739/tkg/management
+```
+
+### Build and Ship Packages to the Registry
+
+Then build and publish all packages according to
+the [development workflow](https://github.com/vmware-tanzu/tanzu-framework/blob/main/docs/packages/dev-workflow.md).
+
+```shell
+make build-cli && make docker-build && make docker-publish && make kbld-image-replace
+```
+
+* `build-cli` compiles the tanzu CLI and its plugins
+* `docker-build` builds container images for all packages
+* `docker-publish` uploads the container images you built to the OCI registry. Make sure you have authenticated the
+  docker per [here](https://cloud.google.com/container-registry/docs/advanced-authentication#gcloud-helper).
+* `local-registry` starts a local registry at `localhost:5001`
+* `kbld-image-replace` replaces `packages/*/kbld-config.yaml` with the resolved image paths
+
+Note: in theory `make docker-all` will trigger all `docker-build`, `docker-publish` and `kbld-image-replace`. But
+locally for me, it only runs the first target with this version
+of [Makefile](https://github.com/vmware-tanzu/tanzu-framework/blob/0f572c2ee0d209f538d33b3c5951a664f4076a82/Makefile#L740)
+.
+
+After that, run the local registry. Then, build the package and repo bundles for a management clusters.
+
+```shell
+make local-registry && make package-push-bundles-repo PACKAGE_REPOSITORY=management
+```
+
+## Link clusters to the package repository and install packages
+### Spin up a kind cluster as the management cluster
+
+Create kind cluster with CAPx providers and kapp-controller installed. Since this utility script install multiple
+infrastructure providers,
+environment variables should be exported such as `AWS_B64ENCODED_CREDENTIALS`, `VSPHERE_PASSWORD`, `VSPHERE_USERNAME`.
+
+```shell
+export AWS_B64ENCODED_CREDENTIALS=<encoded aws credentials>
+./hack/kind/deploy_kind_with_capi_and_kapp.sh
+```
+
+Login to the kind cluster with tanzu CLI.
+
+```shell
+tanzu login --kubeconfig '${HOME}/.kube/config' --context kind-test-cluster --name kind-test-cluster
```

Comment:
the build command above did not build the tanzu command line utility for me, it created the individual fields such as tanzu-package etc. IS this correct? If tanzu is being built, can we point to the correct location?

Reply:
the `tanzu` cli is built when we are running `make build-cli`. please refer to the previous section "Build and Ship Packages to the Registry"